astropy/io/fits/card.py

```diff
@@ -321,7 +321,11 @@ def value(self, value):
                            np.bool_)):
             raise ValueError(f'Illegal value: {value!r}.')
 
-        if isinstance(value, float) and (np.isnan(value) or np.isinf(value)):
+        if (isinstance(value, float) or
+            isinstance(value, np.float32)) and (np.isnan(value) or
+                                                np.isinf(value)):
+            # value is checked for both float and np.float32 instances
+            # since np.float32 is not considered a Python float.
```

Comment:
Please use `isinstance(value, numbers.Real)` instead.

Reply:
Done. Added the test as well and tested again.

Reply:
Add `import numbers` at the top (https://docs.python.org/3/library/numbers.html).